modules/dnn/src/layers/convolution_layer.cpp

```diff
@@ -1210,6 +1218,158 @@ class ConvolutionLayerImpl CV_FINAL : public BaseConvolutionLayerImpl
+
+        auto input_wrapper = inputs[0].dynamicCast<CUDABackendWrapperFP32>();
+        auto input = input_wrapper->getView();
+
+        if (!transformedInput.empty())
+        {
+            inputTransformer.transform(input, transformedInput);
+            input = csl::TensorView<float>(transformedInput);
+        }
+
+        auto output_wrapper = outputs[0].dynamicCast<CUDABackendWrapperFP32>();
+        auto output = output_wrapper->getSpan();
+
+        convoluter.convolve(output, input, filtersTensor, workspace);
+        if (hasBias() || fusedBias)
+            csl::tensor_ops::add<float>(cudnnHandle, 1.0, output, 1.0, biasTensor);
+    }
+
+    void initCUDA(
+        csl::Stream stream,
+        csl::cublas::Handle cublas_handle,
+        csl::cudnn::Handle cudnn_handle,
+        std::size_t& scratch_mem_in_bytes,
+        const std::vector<Ptr<BackendWrapper>>& inputs
+    ) override
+    {
+        cudnnHandle = std::move(cudnn_handle);
+
+        auto input_wrapper = inputs[0].dynamicCast<CUDABackendWrapperFP32>();
+        auto input_shape = input_wrapper->getShape();
+
+        /* 1d, 2d, 3d convolutions are supported */
+        CV_Assert(input_shape.size() >= 3 || input_shape.size() <= 5);
+
+        CV_Assert(blobs.size() >= 1);
+        const auto& filtersMat = blobs[0];
+
+        const auto rank = input_shape.size();
+        const auto output_feature_maps = filtersMat.size[0];
+        const auto input_feature_maps = input_shape[1];
+        const auto input_feature_maps_per_group = filtersMat.size[1];
+        const auto groups = input_feature_maps / input_feature_maps_per_group;
+        CV_Assert(input_feature_maps % input_feature_maps_per_group == 0);
+
+        const Mat& filterWeightsSource = newWeightAndBias ? weightsMat : filtersMat;
+        filtersTensor = createTensorHeaderFromMat(filterWeightsSource);
+        copyMatToTensor<float>(filtersTensor, filterWeightsSource, stream);
+
+        if (hasBias() || fusedBias)
+        {
+            std::vector<int> biasShape(rank, 1);
+            biasShape[1] = output_feature_maps;
+            Mat biasMat(rank, biasShape.data(), CV_32F, &biasvec[0]);
+            biasTensor = createTensorHeaderFromMat(biasMat);
+            copyMatToTensor<float>(biasTensor, biasMat, stream);
+        }
+
+        /* left and right are misleading as the padding is applicable for any number of dimensions
+         * but we use those identifiers to avoid confusion with `pads_begin` and `pads_end`
+         */
+        std::vector<std::size_t> common_padding(rank, 0);
+        std::vector<std::size_t> padding_left(rank, 0), padding_right(rank, 0);
+        if (padMode.empty())
+        {
+            for (int i = 2; i < common_padding.size(); i++)
+            {
+                common_padding[i] = std::min(pads_begin[i - 2], pads_end[i - 2]);
+                padding_left[i] = pads_begin[i - 2] - common_padding[i];
+                padding_right[i] = pads_end[i - 2] - common_padding[i];
```

Comment:
The CUDA backend supports asymmetric padding but the OpenCV backend doesn't. For some reason, it throws an error for asymmetric padding in the layer constructor. Hence, despite supporting asymmetric padding, it's not possible to be used.

Reply:
Paddings in some configurations do not make any difference to the output. If the asymmetric padding does not affect the output, it must be ignored.

[TODO]